lib/pure/os.nim

```diff
@@ -1591,13 +1594,15 @@ iterator walkDirRec*(dir: string, yieldFilter = {pcFile},
   ## ``pcLinkToDir``         follow symbolic links to directories
   ## ---------------------   ---------------------------------------------
   ##
-  var stack = @[dir]
+  var stack = @[(fullPath: dir, relPath: "")]
```

Comment:
Seems easier to only push the ``dir``, operate on relative dirs and only in the ``yield`` add the ``dir`` to turn it into an absolute path.

Reply:
Done